wis2box_api/plugins/process/csv_publish.py

```diff
@@ -0,0 +1,192 @@
+###############################################################################
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+###############################################################################
+
+import logging
+
+from pygeoapi.process.base import BaseProcessor
+
+from wis2box_api.wis2box.handle import handle_error
+from wis2box_api.wis2box.handle import DataHandler
+from wis2box_api.wis2box.station import Stations
+from wis2box_api.wis2box.csv_templates import default_mappings
+
+from csv2bufr import transform as transform_csv
+
+from pathlib import Path
+
+import json
+
+LOGGER = logging.getLogger(__name__)
+
+PROCESS_METADATA = {
+    'version': '0.1.0',
+    'id': 'wis2box-csv-process',
+    'title': 'Process and publish CSV from Automatic Weather Stations',
+    'description': 'Converts the posted data to BUFR and publishes to specified topic',  # noqa
+    'keywords': [],
+    'links': [],
+    'inputs': {
+        'channel': {
+            'title': {'en': 'Channel'},
+            'description': {'en': 'Channel / topic to publish on'},
+            'schema': {'type': 'string', 'default': None},
+            'minOccurs': 1,
+            'maxOccurs': 1,
+            'metadata': None,
+            'keywords': []
+        },
+        "data": {
+            "title": "CSV Data",
+            "description": "Input CSV data",
+            "schema": {"type": "string"},
+            "minOccurs": 1,
+            "maxOccurs": 1,
+            "metadata": None,
+            "keywords": [],
+        },
+        "template": {
+            "title": "Mapping",
+            "description": "Mapping-template for CSV to BUFR conversion",
+            "schema": {"type": "string"},
+            "minOccurs": 1,
+            "maxOccurs": 1,
+            "metadata": None,
+            "keywords": [],
+        },
+        "notify": {
+            "title": "Notify",
+            "description": "Enable WIS2 notifications",
+            "schema": {"type": "boolean"},
+            "minOccurs": 1,
+            "maxOccurs": 1,
+            "metadata": None,
+            "default": True
+        }
+    },
+    'outputs': {
```

Comment:
This appears to be different to that used in the tests, is this intended?

Reply:
Yes I wanted one test returning raw data and another test returning file-urls , so I put notify=false to one of the tests.